src/components/SessionTimeout/SessionTimeoutModalContainer.js

```diff
@@ -0,0 +1,32 @@
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router';
+import SessionTimeoutModal from './SessionTimeoutModal';
+import { sessionTimedOut } from '../../actions/SessionTimeout/sessionTimeoutAction';
+
+const mapStateToProps = (state, props) =>
+    ({
+        sessionTimedOut: state.sessionTimedOut
+    });
+
+const mapDispatchToProps = (dispatch, props) =>
+    ({
+        startTimer(){
+            startTimer(dispatch);
+        },
+        restartTimer() {
+            dispatch(sessionTimedOut(false));
+            startTimer(dispatch);
+        },
+        sessionTimedOut(isTimedOut) {
+            dispatch(sessionTimedOut(isTimedOut));
+        }
+    });
+
+export default withRouter(connect(mapStateToProps, mapDispatchToProps)(SessionTimeoutModal));
+
+export const startTimer = (dispatch) => {
+    clearTimeout();
+    setTimeout(function() {
+        dispatch(sessionTimedOut(true));
```

Comment:
This also seemed funky to me, dispatching a function that dispatches the change. I might be missing something, could it instead just be `sessionTimedOut(true)`?

Reply:
Oh, I see the confusion here. So, the dispatch is going to call the sessionTimedOut imported at the top of the file, not call the sessionTimedOut function that is defined in mapDispatchToProps. So, it is sending the action directly to React to handle the state updates.